src/models/model.cpp

```diff
@@ -480,6 +480,23 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
       }
 
       session_options.AppendExecutionProvider_CUDA_V2(*ort_provider_options);
+    } else if (provider_options.name == "nv") {
+      std::vector<const char*> keys, values;
+      for (auto& option : provider_options.options) {
+        keys.emplace_back(option.first.c_str());
+        values.emplace_back(option.second.c_str());
+      }
+
+      auto ort_provider_options = OrtNvProviderOptions::Create();
+      // TODO: missing handling in ORT-> NV-EP, need to verify and enable.
+      //ort_provider_options->Update(keys.data(), values.data(), keys.size());
+      session_options.AppendExecutionProvider_Nv(*ort_provider_options);
+
+      // Enable CUDA EP for contrib ops
+      auto ort_provider_options_cuda = OrtCUDAProviderOptionsV2::Create();
+      ort_provider_options_cuda->Update(keys.data(), values.data(), keys.size());
+
+      session_options.AppendExecutionProvider_CUDA_V2(*ort_provider_options_cuda);
```

Comment:
Why isn't this just using the existing session_options.AppendExecutionProvider API? Then it's just passing "nv" as the provider name to the one function, the keys/values are shared with all provider types there.

I see the change in Ort, I'll leave a comment there too.